src/runtime/manifest-parser.peg

```diff
@@ -247,37 +273,13 @@ Interface "an interface"
     });
   }
 
-InterfaceItem
-  = InterfaceSlot
-  / InterfaceArgument
-
 InterfaceArgument
-  = name:NameWithColon direction:Direction? type:(whiteSpace? ParticleHandleConnectionType)? eolWhiteSpace
-  { return buildInterfaceArgument(name, direction || 'any', optional(type, t => t[1], null)); }
-  / direction:Direction type:(whiteSpace? ParticleHandleConnectionType)? eolWhiteSpace
-  { return buildInterfaceArgument('*', direction || 'any', optional(type, t => t[1], null)); }
-  / type:(whiteSpace? ParticleHandleConnectionType) eolWhiteSpace
-  { return buildInterfaceArgument('*', 'any', type); }
-
-InterfaceSlot
-  = name:NameWithColon? direction:('consumes' / 'provides') isOptional:'?'? type:(whiteSpace SlandleType)? eolWhiteSpace
-  {
-    let isSet = false;
-    if (type) {
-      type = type[1]; // remove the whitespace.
-      isSet = type.isSet;
-      type.fields.forEach(({name, value}) => {
-        error(`interface slots do not currently support fields`);
-      });
-    }
-    return toAstNode<AstNode.InterfaceSlot>({
-      kind: 'interface-slot',
-      name,
-      isRequired: !isOptional,
-      direction,
-      isSet,
-    });
-  }
+  = name:NameWithColon direction:(Direction / SlotDirection)? isOptional:'?'? type:(whiteSpace? ParticleHandleConnectionType)? eolWhiteSpace
+  { return buildInterfaceArgument(name, direction || 'any', isOptional, optional(type, t => t[1], null)); }
+  / direction:(Direction / SlotDirection) isOptional:'?'? type:(whiteSpace? ParticleHandleConnectionType)? eolWhiteSpace
+  { return buildInterfaceArgument(null, direction || 'any', isOptional, optional(type, t => t[1], null)); }
+  / isOptional:'?'? type:(whiteSpace? ParticleHandleConnectionType) eolWhiteSpace
```

Comment:
This looks odd - what's the optionality `?` applying to? Should this just be the type capture on its own?

Reply:
I'd prefer this, but wanted to make sure that specifying an optional 'any' handle was possible.